### ISSUE TITLE
BUG: Timedelta.microseconds is rounded up if nanos>500

### ISSUE THREAD

**Issue by jonyii** [pandas-dev/pandas#33915]
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

**Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

#### Code Sample, a copy-pastable example

```python

t = pd.Timedelta(501, unit='ns')
t.components
print("sec=%d, micro=%d, nano=%d"  % (t.seconds, t.microseconds, t.nanoseconds))

assert t.microseconds == 0, "This should be zero, right?"
```

#### Problem description

Timedelta.microseconds is rounded up based on nanos

#### Expected Output

 t.microseconds == 0

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : None
python           : 3.6.10.final.0
python-bits      : 64
OS               : Linux
OS-release       : 4.14.138+
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : en_US.UTF-8
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.0.3
numpy            : 1.18.1
pytz             : 2019.3
dateutil         : 2.8.1
pip              : 20.0.2
setuptools       : 45.1.0.post20200119
Cython           : 0.29.16
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.11.0
IPython          : 7.13.0
pandas_datareader: None
bs4              : None
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : None
matplotlib       : 3.2.1
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
pytest           : None
pyxlsb           : None
s3fs             : None
scipy            : 1.4.1
sqlalchemy       : 1.3.13
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
xlsxwriter       : None
numba            : None

</details>

**Comment by jonyii** [comment]
the rounding is not consistent either:
```python
for nanos in [-501, 500, 501, 1499, 1500]:
    t = pd.Timedelta(nanos, unit='ns')
    print(f"input nanos={nanos}:\t  t.microseconds = {t.microseconds} t.nanoseconds = {t.nanoseconds}\t", t)
```

![image](https://user-images.githubusercontent.com/12170567/80817258-665cb980-8bd1-11ea-8480-54e700d3cb16.png)

**Comment by mroeschke** [comment]
Thanks for the report. This is solved by https://github.com/pandas-dev/pandas/pull/31380 and will be included in the 1.1 release (in a couple of months)
